### PR TITLE
Add methods to tell if address is in USA to Simple Forms models 

### PR DESCRIPTION
**Pull request by Thrillberg**
## Summary
This PR updates the Simple Forms form models so that they have a method, `zip_code_is_us_based`, which can tell the metadata validator if the address is based in the USA. Up until now, we've been operating under the assumption that all addresses are US-based but that isn't always going to be the case.

## Related issue(s)
https://app.zenhub.com/workspaces/vagov-product-team-forms-634853151f5c6000165942bc/issues/gh/department-of-veterans-affairs/va.gov-team/72916

## Testing done

 Where logic beyond accessing a datapoint from the form data exists, I've added tests, too.

### CODE REVIEW

File: modules/simple_forms_api/spec/models/vba_20_10207_spec.rb [hunk 1]
```diff
@@ -3,6 +3,48 @@
 require 'rails_helper'
 
 RSpec.describe 'SimpleFormsApi::VBA2010207' do
+  describe 'zip_code_is_us_based' do
+    describe 'veteran address is present and in US' do
```
Comment (pennja) [inline modules/simple_forms_api/spec/models/vba_20_10207_spec.rb:7]: The could be `context` calls instead of `describe` calls.

File: modules/simple_forms_api/spec/models/vba_20_10207_spec.rb [hunk 2]
```diff
@@ -3,6 +3,48 @@
 require 'rails_helper'
 
 RSpec.describe 'SimpleFormsApi::VBA2010207' do
+  describe 'zip_code_is_us_based' do
```
Comment (pennja) [inline modules/simple_forms_api/spec/models/vba_20_10207_spec.rb:6]: Is it valuable to include a test or tests that give the context where none of these addresses are given? In other words, are we throwing errors or otherwise catching this?

File: modules/simple_forms_api/spec/models/vba_20_10207_spec.rb [hunk 3]
```diff
@@ -3,6 +3,48 @@
...
+
+        expect(form.zip_code_is_us_based).to eq(true)
+      end
+    end
+
+    describe 'veteran address is present and not in US' do
+      it 'returns false' do
+        data = { 'veteran_mailing_address' => { 'country' => 'Canada' } }
+
+        form = SimpleFormsApi::VBA2010207.new(data)
```
[the remark this answers is not in the document]
Comment (pennja) [reply modules/simple_forms_api/spec/models/vba_20_10207_spec.rb:21]: You could even go a step further and make it the `subject` of the test:

```suggestion
        subject(:zip_code_is_us_based) { described_class.new(data).zip_code_is_us_based }
```

Then you could do something like this within each context:

```ruby
  let(:data) { { 'veteran_mailing_address' => { 'country' => 'Canada' } } }

  it { is_expected.to return_falsey }
```

I THINK that should work, may need to tweak it.

File: modules/simple_forms_api/spec/models/vba_20_10207_spec.rb (hunk 2 above)
Comment (Thrillberg) [reply modules/simple_forms_api/spec/models/vba_20_10207_spec.rb:6]: Great point! I will add those! 👍

File: modules/simple_forms_api/app/models/simple_forms_api/vba_40_10007.rb [hunk 4]
```diff
@@ -24,6 +24,11 @@ def metadata
       }
     end
 
+    def zip_code_is_us_based
+      # TODO: Implement this
+      true
+    end
```
Comment (pennja) [inline modules/simple_forms_api/app/models/simple_forms_api/vba_40_10007.rb:30]: Will these methods be implemented in a future PR? If so, might I suggest creating a class that has this method defined and pointing this and all other classes utilizing this method to inherit from it, like so:

```ruby
class BaseVbaForm
  def zip_code_is_us_based
    true
  end
end
```

```ruby
module SimpleFormsApi
  class VBA210845 << BaseVbaForm
    ...
  end
end
```

Perhaps I'm overthinking things but any other shared logic can live there too. I'd love your thoughts on this.

**Review by pennja — APPROVED** [review]
I left one suggestion about inheritance. It would also guard against any new forms being created that failed to define that method. Just food for thought. Otherwise, this looks great. Good job! 👍

File: modules/simple_forms_api/app/models/simple_forms_api/vba_40_10007.rb (hunk 4 above)
Comment (Thrillberg) [reply modules/simple_forms_api/app/models/simple_forms_api/vba_40_10007.rb:30]: Oh yeah that's a great idea! I hesitate to create an abstraction out of this but maybe that day is soon approaching. I think for now I'll keep this as-is but maybe when the form models are more stable we can look at them and abstract some of their methods into a parent class.

**Review by pennja — APPROVED** [review]
Looks good, great work! 👍
